core/domain/commit_commands_domain_test.py

```diff
@@ -0,0 +1,279 @@
+        name: str. The command name.
+        parameters: dict. Dict containing command parameters.
+    """
+    commit_cmd_domain_object = commit_cmd_class(
+        name=name, parameters=parameters)
+    commit_cmd_domain_object.validate()
+
+
+def validate_with_missing_keys_in_command(
+        self, commit_cmd_class, name, parameters):
+    """Validates a commit command with missing keys in parameters.
+
+    Args:
+        commit_cmd_class: CommitCmdDomainClass. The domain class of the command
+            to validate.
+        name: str. The command name.
+        parameters: dict. Dict containing command parameters.
+    """
+
+    if not len(parameters.keys()):
+        return
+    parameter_key = sorted(parameters.keys())[0]
+    parameters.pop(parameter_key, None)
+    commit_cmd_domain_object = commit_cmd_class(
+        name=name, parameters=parameters)
+    assert_validation_error(
+        self, commit_cmd_domain_object,
+        'Following required keys are missing: %s' % parameter_key)
+
+
+def validate_with_extra_keys_in_command(
+        self, commit_cmd_class, name, parameters):
+    """Validates a commit command with extra keys in parameters.
+
+    Args:
+        commit_cmd_class: CommitCmdDomainClass. The domain class of the command
+            to validate.
+        name: str. The command name.
+        parameters: dict. Dict containing command parameters.
+    """
+    parameters['random'] = 'random'
+    commit_cmd_domain_object = commit_cmd_class(
+        name=name, parameters=parameters)
+    assert_validation_error(
+        self, commit_cmd_domain_object,
+        'Following extra keys are present: random')
+
+
+class ExplorationCommitCmdUnitTests(test_utils.GenericTestBase):
+    """Test the exploration commit cmd domain object."""
+
+    def test_commands(self):
+        command_list = [{
+            'name': exp_domain.CMD_ADD_STATE,
+            'parameters': {'state_name': 'state'}
+        }, {
+            'name': exp_domain.CMD_DELETE_STATE,
+            'parameters': {'state_name': 'state'}
+        }, {
+            'name': exp_domain.CMD_RENAME_STATE,
+            'parameters': {
+                'old_state_name': 'old_state', 'new_state_name': 'new_state'
+            }
+        }, {
+            'name': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'parameters': {
+                'property_name': 'property', 'state_name': 'state',
+                'new_value': 'value'
+            }
+        }, {
+            'name': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+            'parameters': {
+                'property_name': 'property', 'new_value': 'new_value',
+                'old_value': 'old_value'
+            }
+        }, {
+            'name': exp_domain.CMD_MIGRATE_STATES_SCHEMA_TO_LATEST_VERSION,
+            'parameters': {'from_version': 1, 'to_version': 3}
+        }]
+
+        commit_cmd_class = commit_commands_domain.ExplorationCommitCmd
+        for cmd in command_list:
```

Comment:
Add a function to iterate over the command list and do the validation tests instead of having the loop multiple times.

Reply:
Done.